pkg/controller/daemon/daemoncontroller.go

```diff
@@ -510,6 +527,10 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	if err != nil {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
+	// Check DeletionTimestamp again because it may have been refreshed by getNodesToDaemonPods().
+	if ds.DeletionTimestamp != nil {
```

Comment:
@kargakis The problem with fetching the live object once at the beginning is that we might encounter the following:

1. Fetch live controller object. DeletionTimestamp = nil
2. GC sets DeletionTimestamp != nil, and orphans one or more Pods.
3. Get list of Pods. See some orphans. Readopt.

@caesarxuchao suggested that the rule to prevent this is that you must refetch the controller object some time after you get the list of Pods. Maybe we could move the Pod list fetch to the top as well?